benchmark_test.go

```diff
@@ -218,3 +220,88 @@ func TestBenchmarkStatsPerPeer(t *testing.T) {
 	})
 	assert.Equal(t, want, statsServer.Aggregated(), "unexpected stats")
 }
+
+func TestBenchmarkOutput(t *testing.T) {
+	// Testing both plaintext and JSON output
+	tests := []struct {
+		name          string
+		opts          BenchmarkOptions
+		wantOutput    []string
+		notWantOutput []string
+	}{
+		{
+			name: "json",
+			opts: BenchmarkOptions{
+				Format: "json",
+			},
+			wantOutput:    []string{"summary", "benchmarkParameters", "maxRPS", "latencies"},
+			notWantOutput: []string{"Errors", "Benchmark parameters", "Max RPS"},
+		},
+		{
+			name: "plaintext",
+			opts: BenchmarkOptions{
+				Format: "plaintext",
+			},
+			wantOutput:    []string{"Benchmark parameters", "Max RPS"},
+			notWantOutput: []string{"Errors", "summary", "maxRPS"},
+		},
+		// Unimplemented format should default to plaintext
+		{
+			name: "unrecognized",
+			opts: BenchmarkOptions{
+				Format: "csv",
+			},
+			wantOutput:    []string{"Benchmark parameters", "Max RPS"},
+			notWantOutput: []string{"Errors", "benchmarkParameters", "maxRPS"},
+		},
+	}
+
+	var requests atomic.Int32
+	s := newServer(t)
+	defer s.shutdown()
+	s.register(fooMethod, methods.errorIf(func() bool {
+		requests.Inc()
+		return false
+	}))
+	m := benchmarkMethodForTest(t, fooMethod, transport.TChannel)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			requests.Store(0)
+			buf, _, out := getOutput(t)
+			opts := Options{
+				BOpts: BenchmarkOptions{
+					MaxRequests: 10,
+					MaxDuration: 200 * time.Millisecond,
+					RPS:         120,
+					Connections: 50,
+					Concurrency: 2,
+					Format:      tt.opts.Format,
+				},
+				TOpts: s.transportOpts(),
+			}
+
+			runBenchmark(out, _testLogger, opts, _resolvedTChannelThrift, m)
+			bufStr := buf.String()
+
+			for _, want := range tt.wantOutput {
+				assert.Contains(t, bufStr, want)
+			}
+			for _, notWant := range tt.notWantOutput {
+				assert.NotContains(t, bufStr, notWant)
+			}
+
+			if opts.BOpts.Format == "json" {
```

Comment:
optional: if you want to test all cases of json, e.g. (json, JSON, Json, etc), you can add an additional field in the test struct, e.g. `wantJSON bool`, and check it here